src/server/services/procedures/historical-temperature/historical-temperature.js

```diff
@@ -0,0 +1,149 @@
+        return 'Unknown country or region';
+    }
+};
+
+/*
+ * Associates less obviously named regions to their URLs
+ */
+const regionsDictionary = {
+    allland: 'Global/Complete_TAVG_complete.txt',
+    'all-land': 'Global/Complete_TAVG_complete.txt',
+    global: 'Global/Land_and_Ocean_complete.txt',
+    northern: 'Regional/TAVG/Text/northern-hemisphere-TAVG-Trend.txt',
+    southern: 'Regional/TAVG/Text/southern-hemisphere-TAVG-Trend.txt',
+    'northern-hemisphere': 'Regional/TAVG/Text/northern-hemisphere-TAVG-Trend.txt',
+    'southern-hemisphere': 'Regional/TAVG/Text/southern-hemisphere-TAVG-Trend.txt',
+};
+
+/**
+ * Get the monthly anomaly data for a region
+ * @param {String} region Name of region/country
+ * @returns {Array<Array>} Monthly data points
+ */
+BerkeleyEarth.monthlyAnomaly = function (region) {
+    return this._getCountryData(region, 'monthly');
+};
+
+/**
+ * Get the 12 month averaged anomaly data for a region
+ * @param {String} region Name of region/country
+ * @returns {Array<Array>} Monthly data points
+ */
+BerkeleyEarth.annualAnomaly = function (region) {
+    return this._getCountryData(region, 'annual');
+};
+
+/**
+ * Get the 5-year averaged anomaly data for a region
+ * @param {String} region Name of region/country
+ * @returns {Array<Array>} Monthly data points
+ */
+BerkeleyEarth.fiveYearAnomaly = function (region) {
+    return this._getCountryData(region, '5year');
+};
+
+/**
+ * Get the 10-year averaged anomaly data for a region
+ * @param {String} region Name of region/country
+ * @returns {Array<Array>} Monthly data points
+ */
+BerkeleyEarth.tenYearAnomaly = function (region) {
+    return this._getCountryData(region, '10year');
+};
+
+/**
+ * Get the 20-year averaged anomaly data for a region
+ * @param {String} region Name of region/country
+ * @returns {Array<Array>} Monthly data points
+ */
+BerkeleyEarth.twentyYearAnomaly = function (region) {
+    return this._getCountryData(region, '20year');
+};
+
+// Indices for data columns
+BerkeleyEarth._dataColumns = {
+    'monthly': 2,
+    'annual': 4,
+    '5year': 6,
+    '10year': 8,
+    '20year': 10,
+};
+
+/**
+ * Requests the data for a country/region
+ * @param {String} country Country/region to get data for
+ * @param {String} type Type of data to return
+ * @returns {Array<Array>} Parsed data
+ */
+BerkeleyEarth._getCountryData = function (country, type) {
+    country = country.toLowerCase().trim().replace(/\s+/, '-');
+    const options = {
+        path: `Regional/TAVG/Text/${country}-TAVG-Trend.txt`
+    };
+    // Check for special region names
+    if (Object.keys(regionsDictionary).indexOf(country) !== -1) {
+        options.path = regionsDictionary[country];
+    }
+    return this._requestData(options).then(this._extractData.bind(this, type)).catch(err => {
+        const prettyError = this._rewordError(err);
+        if (prettyError) {
+            return this.response.status(500).send(prettyError);
```

Comment:
As errors thrown by RPCs will be automatically handled by NetsBlox, it is preferred to just throw errors with meaningful error messages rather than send the response directly from the RPC code. On example can be found in the [BaseX service](https://github.com/NetsBlox/NetsBlox/blob/master/src/server/services/procedures/base-x/base-x.js#L64-L86).

Admittedly, the (no longer supported) IEXTrading service [still uses this outdated approach](https://github.com/NetsBlox/NetsBlox/blob/master/src/server/services/procedures/iex-trading/iex-trading.js#L141-L145) and should be probably updated or removed so contributors don't use it as an example or for reference.

Reply:
The new commit changes this.